xds/src/main/java/io/grpc/xds/SharedCallCounterMap.java

```diff
@@ -73,6 +73,9 @@ void cleanQueue() {
     CounterReference ref;
     while ((ref = (CounterReference) refQueue.poll()) != null) {
       Map<String, CounterReference> clusterCounter = counters.get(ref.cluster);
```

Comment:
I'm guessing `clusterCounter` shouldn't be null, because refs should be enqueued in the same order as the order of  underlying referents being nullified by garbage collector. But I did not see javadoc explicitly say that. 

Is there any risk of NPE in extreme race case like the following?
ref1.referent nullified by => ref2 created and put in the counters map => ref2.referent nullified by gc => ref2 enqueued => ref1 enqueued.

Reply:
Done. TIL thanks.